runtime/compiler/control/J9Options.cpp

```diff
@@ -2304,6 +2304,18 @@ J9::Options::fePreProcess(void * base)
    if (FIND_ARG_IN_VMARGS(EXACT_MATCH, "-Xnoclassgc", 0) >= 0)
       self()->setOption(TR_NoClassGC);
 
+   // fePreProcess is called twice - for AOT and JIT options parsing, which is redundant in terms of
+   // processing the -XperfTool option. AOT parsing happens first and then JIT, hence the boolean
+   // check to avoid setting TR_PerfTool during AOT parsing.
+   // We should set TR_PerfTool only once for JIT though to avoid duplicate messages.
+   static bool perfToolNotYetParsed = true;
+   if (!perfToolNotYetParsed)
+      {
+      if (FIND_ARG_IN_VMARGS(EXACT_MATCH, "-XperfTool", 0) >= 0)
+         self()->setOption(TR_PerfTool);
+      }
+   perfToolNotYetParsed = false;
```

Comment:
If the AOT option is preprocessed first then as the code stands, the JIT won't realize the option has been set. This is because 
https://github.com/eclipse-openj9/openj9/blob/7744616e477b18ce50a28150d431a136c0b3f04d/runtime/compiler/control/CompilationThread.cpp#L7932-L7936
only checks the JIT options.

From what I can tell, there's other places similar to this that only check the JIT cmdline option. The easiest solution is to just let it get parsed twice. Another approach is to do this check in `onLoadInternal` after we've called `processOptionsJIT` and then set it on the JIT options, ie:
```
if (FIND_ARG_IN_VMARGS(EXACT_MATCH, "-XperfTool", 0) >= 0)
   TR::Options::getCmdLineOptions()->setOption(TR_PerfTool);
```

Reply:
**If the AOT option is preprocessed first then as the code stands, the JIT won't realize the option has been set**. ---> Which is why, I'm not setting in the first iteration **perfToolNotYetParsed** is set true initially and am checking with (!perfToolNotYetParsed) . The TR_PerfTool will be set in the second iteration(for JIT). Checked the working of this code and it is getting set for JIT and perf-%pid map file is getting generated.

**The easiest solution is to just let it get parsed twice** --> Initially I've done the same, but the stderr message below is getting printed twice as `latePostProcess` is called from `latePostProcessJIT` & `latePostProcessAOT` which is why I'm setting it only once in second iteration for JIT.

https://github.com/eclipse-openj9/openj9-omr/blob/16e09fb2616d52eb2cffe51c1b9a40d7b23fcb50/compiler/control/OMROptions.cpp#L2033-L2038

Reply:
**Another approach is to do this check in onLoadInternal after we've called processOptionsJIT and then set it on the JIT options** --> I'll try this. Thanks @dsouzai